cmd/nerdctl/compose_up_linux_test.go

```diff
@@ -437,3 +437,54 @@ services:
 	base.ComposeCmd("-f", comp.YAMLFullPath(), "down").AssertOK()
 
 }
+
+func TestComposeUpWithExternalNetwork(t *testing.T) {
+	containerName1 := "nginx-1"
+	containerName2 := "nginx-2"
+	var dockerComposeYaml1 = fmt.Sprintf(`
+version: "3"
+services:
+  nginx-1:
+    image: %s
+    container_name: %s
+    networks:
+      test:
+        aliases:
+          - nginx-1
+networks:
+  test:
+    external: true
+`, testutil.NginxAlpineImage, containerName1)
+	var dockerComposeYaml2 = fmt.Sprintf(`
+version: "3"
+services:
+  nginx-2:
+    image: %s
+    container_name: %s
+    networks:
+      test:
+        aliases:
+          - nginx-2
+networks:
+  test:
+    external: true
+`, testutil.NginxAlpineImage, containerName2)
+	comp1 := testutil.NewComposeDir(t, dockerComposeYaml1)
+	defer comp1.CleanUp()
+	comp2 := testutil.NewComposeDir(t, dockerComposeYaml2)
+	defer comp2.CleanUp()
+	base := testutil.NewBase(t)
+	// Create the test network
+	base.Cmd("network", "create", "test").AssertOK()
```

Comment:
The network name has to be `testutil.Identifier(t)` and has to be removed with `defer`